### PR TITLE
Using react_on_rails gem for rendering components on the client

### PR DESCRIPTION
**Pull request by mapreal19**
- Paired with @samnang

### CODE REVIEW

**Comment by justin808** [comment]
@samnang  @mapreal19 , please let me know if this ready for merging and deployment.

File: app/views/pages/index.html.erb [hunk 1]
```diff
@@ -16,7 +16,6 @@
   </li>
 </ul>
 <hr/>
-<div id="content"></div>
-
-
-
+<div id="content">
```
Comment (justin808) [inline app/views/pages/index.html.erb]: Do we need the wrapper for id "content" still?

File: app/views/pages/index.html.erb [hunk 2]
```diff
@@ -16,7 +16,6 @@
...
+  <%= react_component('App') %>
```
Comment (justin808) [inline app/views/pages/index.html.erb]: OK -- we should either comment or explicitly say that server rendering is off.

I think the next PR should be two have 2 tabs that show with and without server rendering.

@alexfedoseev @samnang @skv-headless @mapreal19 @PlasticLizard thoughts?

Comment (alexfedoseev) [inline app/views/pages/index.html.erb]: I see no reasons why it can't be implemented (:

Comment (samnang) [inline app/views/pages/index.html.erb]: @justin808 I totally agree we will 2 separate tabs for client side rendering and server side rendering. The idea for this PR is for client side rendering.

File: app/views/pages/index.html.erb (hunk 1 above)
Comment (samnang) [inline app/views/pages/index.html.erb]: Actually, we don't need it. But we haven't looked at if we need it for any styling.

**Comment by samnang** [comment]
@justin808 everything is ready and work as what we have before. We've just switched to use the gem instead.

**Comment by justin808** [comment]
Sweet. I'm going to look at this tonight, and probably deploy it!

**Comment by justin808** [comment]
Please continue conversation on #84.

**Pull request closed without merging**